firefox_ui_harness/mixins/base.py

```diff
@@ -39,9 +49,54 @@
     'focusmanager.testmode': True,
     'geo.provider.testing': True,
     'javascript.options.showInConsole': True,
+    'marionette.logging': False,
     'security.notification_enable_delay': 0,
     'signon.rememberSignons': False,
     'startup.homepage_welcome_url': 'about:blank',
     'toolkit.startup.max_resumed_crashes': -1,
     'toolkit.telemetry.enabled': False,
 }
+
+
+class FirefoxUIOptions(BaseMarionetteOptions):
+    def __init__(self, **kwargs):
+        BaseMarionetteOptions.__init__(self, **kwargs)
+
+        # Inheriting object must call this __init__ to set up option handling
+        self.add_option('--installer',
+                        dest='installer',
+                        help='installer of a Gecko application to use for running the tests')
+
+    def parse_args(self, *args, **kwargs):
+        options, tests = BaseMarionetteOptions.parse_args(self, *args, **kwargs)
+
+        # Bug 1142064 - We cannot easily extent options because registered handlers
+        # are called at the end in MarionetteBaseOptions.verify_usage(). As result it
+        # will abort due to no binary specified. Once the bug is fixed we can move
+        # the whole block to self.base_verify_usage().
+        if options.installer:
+            if options.binary:
+                self.error('Options --binary and --installer are mutually exclusive.')
+
+            options.binary = 'FAKED_VALUE'
+
+        if not tests:
+            tests = [firefox_puppeteer.manifest, firefox_ui_tests.manifest_all]
```

Comment:
No we don't want to. Thanks for that hint. We should indeed not add them to the top-level manifest file. Once we have specific entry points for functional and remote tests, I would even remove that top-level manifest completely.